pkg/flowaggregator/clickhouseclient/clickhouseclient_test.go

```diff
@@ -364,3 +331,59 @@ func TestUpdateCH(t *testing.T) {
 		return (err == nil), nil
 	}), "timeout while waiting for second flow record to be committed (after DB connection update)")
 }
+
+func TestParseDatabaseURL(t *testing.T) {
+	testcases := []struct {
+		url           string
+		expectedProto protocol
+		expectedAddr  string
+		expectedErr   error
```

Comment:
you should make this a string since you use `ErrorContains` as an assertion anyway. No reason to build an `error` just to access the message later with `.Error()`